qhub/template/{{ cookiecutter.repo_directory }}/templates/jupyterhub_config.py.j2

```diff
@@ -142,11 +144,17 @@ def qhub_configure_profile(username, safe_username, profile):
     profile['kubespawner_override']['gid'] = primary_gid
     profile['kubespawner_override']['supplemental_gids'] = secondary_gids
     profile['kubespawner_override']['fs_gid'] = primary_gid
+
+    # Cost monitoring labels
+    k8s_safe_chars = set(string.ascii_lowercase + string.digits + '.-_')
+    k8s_safe_username = escapism.escape(username.lower().replace('@', '_'), safe=k8s_safe_chars, escape_char='-')
+    profile['kubespawner_override']['extra_labels'] = {
+        'owner': k8s_safe_username,
+        'team': user['primary_group']
+    }
```

Comment:
Is this intentionally part of this pull request?

Reply:
Well, sort of - they were really part of my Kubecost experiment, but think they could be useful for that and/or any other monitoring systems that people might want to try out in the meantime.

I think JupyterHub already adds another label with its own name for the owner anyway.

So, yes, I did include it on purpose but they aren't actually related to the dashboard auth.

Reply:
OK, makes sense.